frontend/src/components/utilities/attemptLogin.ts

```diff
@@ -22,118 +19,100 @@ interface IsLoginSuccessful {
+  email,
+  password,
+  providerAuthToken
+}: {
+  email: string;
+  password: string;
+  providerAuthToken?: string;
+}): Promise<IsLoginSuccessful> => {
   const telemetry = new Telemetry().getInstance();
-  return new Promise((resolve, reject) => {
-    client.init(
-      {
-        username: email,
-        password
-      },
-      async () => {
-        try {
-          const clientPublicKey = client.getPublicKey();
-          
-          const { serverPublicKey, salt } = await login1({
-            email,
-            clientPublicKey,
-            providerAuthToken,
-          });
-          
-          client.setSalt(salt);
-          client.setServerPublicKey(serverPublicKey);
-          const clientProof = client.getProof(); // called M1
-          
-          const {
-            mfaEnabled,
-            encryptionVersion,
-            protectedKey,
-            protectedKeyIV,
-            protectedKeyTag,
-            token, 
-            publicKey, 
-            encryptedPrivateKey, 
-            iv, 
-            tag 
-          } = await login2(
-            {
-              email,
-              clientProof,
-              providerAuthToken,
-            }
-          );
-          
-          if (mfaEnabled) {
-            // case: MFA is enabled
-
-            // set temporary (MFA) JWT token
-            SecurityClient.setMfaToken(token);
-
-            resolve({
-              mfaEnabled,
-              success: true
-            });
-          } else if (
-            !mfaEnabled &&
-            encryptionVersion &&
-            encryptedPrivateKey &&
-            iv &&
-            tag &&
-            token
-          ) {
-            // case: MFA is not enabled
-
-            // unset provider auth token in case it was used
-            SecurityClient.setProviderAuthToken("");            
-            // set JWT token
-            SecurityClient.setToken(token);
-            
-            const privateKey = await KeyService.decryptPrivateKey({
-              encryptionVersion,
-              encryptedPrivateKey,
-              iv,
-              tag,
-              password,
-              salt,
-              protectedKey,
-              protectedKeyIV,
-              protectedKeyTag
-            });
-
-            saveTokenToLocalStorage({
-              publicKey,
-              encryptedPrivateKey,
-              iv,
-              tag,
-              privateKey
-            });
-
-            if (email) {
-              telemetry.identify(email, email);
-              telemetry.capture("User Logged In");
-            }
-            
-            resolve({
-              mfaEnabled: false,
-              success: true
-            });
-          }
-        } catch (err) {
-          reject(err);
-        }
-      }
-    );
+  // eslint-disable-next-line new-cap
+  const client = new jsrp.client();
+  await new Promise((resolve) => {
+    client.init({ username: email, password }, () => resolve(null));
```

Comment:
From the exampels https://github.com/alax/jsrp it shows that the client instances need to be in the call back but you have nothing as the call back. How does this change fix the race condition error?

Reply:
There are two async ways in nodejs. Callbacks was something before promises was introduced. 

Here the callbacks is used, to run async code or the async client when its ready. Instead we use promise so that the client init is done and we wait until its done making the client instance ready to use

The issue which is my still hypothesis is the mix of callback and async try catch  inside a callback giving unexpected behavoirs.